### PR TITLE
[test] Add out-of-core MLPerf training benchmark

### PR DESCRIPTION
**Pull request by stephanie-wang**
Signed-off-by: Stephanie Wang <swang@cs.berkeley.edu>

## Why are these changes needed?

Adds release test for out-of-core dataset ingest in MLPerf. This tests three configurations, using the same file size for each:
1. fits in memory
2. requires spilling to disk
3. should result in OutOfDiskError

### CODE REVIEW

**Comment by stephanie-wang** [comment]
Test run [here](https://buildkite.com/ray-project/release-tests-pr/builds/19327#01841a2b-b8f9-417c-839c-08f5b5bd02b1).

Currently hangs on file size=512 due to a bug in Ray core when out of disk space. We can update the test once this bug is fixed.

**Review by scv119 — APPROVED** [review]

**Comment by scv119** [comment]
the test run looks timed out, is it due to the hang or should we increase the timeout?

**Comment by stephanie-wang** [comment]
> the test run looks timed out, is it due to the hang or should we increase the timeout?

I think it's because of the hang! There is no progress in the output for the last run. Also checked the dashboard, and it does seem to stop right around 90% of disk usage. I removed that config from the actual test for now so that it can pass.

File: release/air_tests/air_benchmarks/mlperf-train/out_of_core_benchmark.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+# Test Ray Data bulk ingestion performance as the size of input files change.
```
Comment (cadedaniel) [inline release/air_tests/air_benchmarks/mlperf-train/out_of_core_benchmark.sh:2]: nit: update comment

**Review by cadedaniel — APPROVED** [review]

**Pull request merged**